### PR TITLE
fix(outdated): emit official version schema from registry tags

### PR DESCRIPTION
**Pull request by eve0415**
`cella outdated` was emitting a wrong, non-spec shape — per-feature `{current, latest}` plus an invented `updateAvailable`/`STATUS` — sourced from the GitHub `devcontainer-collection.json` index, which only carries one version per feature and so *structurally can't* produce `wanted`. This rewrites it as a faithful port of the official `loadVersionInfo`.

### What it does now

For each versionable OCI feature ref it lists the registry tags (`/v2/<name>/tags/list`), filters to strict full semver, sorts descending, and emits the five official fields per feature: `current`, `wanted`, `wantedMajor`, `latest`, `latestMajor` (undefined fields omitted, no nulls). Non-versionable refs (local paths, tarball URLs, deprecated bare ids) are filtered out like the official `getRef`, and features come out in config declaration order. `current` reads the lockfile when present; `wanted` is computed purely from the tag (not backfilled from the lockfile). Text output is the 4-column `Feature/Current/Wanted/Latest` table; JSON is `{ "features": { … } }`, compact when piped and pretty on a TTY.

### The semver subtlety

`wanted` needs node-semver range semantics, and Rust's `semver` crate diverges: it parses a bare `"1.2"` as `^1.2` (`>=1.2.0 <2.0.0`), whereas node-semver treats `"1.2"` as `>=1.2.0 <1.3.0`. OCI tags can only be `latest`/`N`/`N.M`/`N.M.P[-pre]`, so `satisfies` is implemented explicitly (X-ranges that exclude prereleases; exact match for full versions) rather than via `VersionReq`. Proven live: `go:1.0` resolves `wanted: 1.0.9` (highest 1.0.x) with `latest: 1.3.4` — caret semantics would wrongly give `1.3.4`.

### Validated

Verified byte-identical against the real `@devcontainers/cli` v0.87.0 (text table, compact + pretty JSON, every tag form, declaration order, local-ref filtering, lockfile pinning). 36 unit/integration tests including a network end-to-end test against `ghcr.io/devcontainers/features/node:1`.

### Known edge-case divergences

Digest-pinned refs (`…@sha256:…`) don't resolve the digest's metadata version (cella's ref parser has no digest variant) — rare; punted. GitHub-shorthand `owner/repo/feature` normalizes to ghcr (cella's convention) rather than treating the first segment as a registry. The config-order recovery reads the raw JSONC CST because `serde_json` lacks `preserve_order` workspace-wide (a proper fix is its own change).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: crates/cella-cli/src/commands/outdated.rs [hunk 1]
```diff
@@ -41,216 +66,956 @@ pub struct OutdatedArgs {
...
+/// trailing run that contains no `/`. So in the last path segment (after the
+/// final `/`, or the whole string if none) we cut at the **first** `:`/`@`.
+/// This makes `…/y@sha256:abc` cut at the `@` (not the digest's `:`), and a
+/// registry port colon (which sits before a `/`) is never cut.
+fn feature_id_without_version(feature_id: &str) -> &str {
+    let segment_start = feature_id.rfind('/').map_or(0, |i| i + 1);
+    feature_id[segment_start..]
+        .find([':', '@'])
+        .map_or(feature_id, |rel| &feature_id[..segment_start + rel])
+}
```
[the remark this answers is not in the document]
Comment (eve0415) [reply crates/cella-cli/src/commands/outdated.rs:438]: This compiles — `[char; N]` implements `Pattern` since Rust 1.71 (MSRV here is 1.95), so `find([':', '@'])` is valid; clippy and the 4720-test suite are green. Left as-is.

File: crates/cella-cli/src/commands/outdated.rs [hunk 2]
```diff
@@ -41,216 +66,956 @@ pub struct OutdatedArgs {
...
+        source_order
+            .iter()
+            .position(|k| k == key)
+            .unwrap_or(usize::MAX)
+    };
+    let mut ordered = features;
+    // Stable sort keeps the original order among any keys missing from the
+    // source (all ranked `usize::MAX`).
+    ordered.sort_by_key(|(key, _)| rank(key));
+    ordered
```
[the remark this answers is not in the document]
Comment (eve0415) [reply crates/cella-cli/src/commands/outdated.rs]: Fixed in 9f6b0c8 — ranks each key once via a map, so the sort is O(n log n).

File: crates/cella-cli/src/commands/outdated.rs [hunk 3]
```diff
@@ -41,216 +66,956 @@ pub struct OutdatedArgs {
...
+        // `extract_features` loses source order (serde_json sorts object keys
+        // without the `preserve_order` feature). The official CLI emits features
+        // in config declaration order, so re-sort by the order they appear in
+        // the raw JSONC.
+        let features = order_features_by_source(&raw, resolve::extract_features(&config));
+
+        // Lockfile is keyed by the full userFeatureId (the reference as written
+        // in the config), matching the official `lockfile.features[id]` lookup.
+        // A corrupt/unreadable lockfile is treated as absent here (best-effort).
+        let lockfile = cella_features::read_lockfile(&config_path).ok().flatten();
```
[the remark this answers is not in the document]
Comment (eve0415) [reply crates/cella-cli/src/commands/outdated.rs]: Fixed in 9f6b0c8 — an unreadable lockfile is now logged (warn) before being treated as absent.

File: crates/cella-cli/src/commands/outdated.rs [hunk 4]
```diff
@@ -41,216 +66,956 @@ pub struct OutdatedArgs {
...
-        }
+        };
 
-        match self.output {
-            OutputFormat::Auto | OutputFormat::Text => print_table(&entries),
-            OutputFormat::Json => print_json(&entries)?,
-        }
+        let lockfile_version = lockfile
+            .and_then(|lf| lf.features.get(user_feature_id))
+            .map(|entry| entry.version.clone());
```
[the remark this answers is not in the document]
Comment (eve0415) [reply crates/cella-cli/src/commands/outdated.rs]: Good catch. Root cause is that cella's lockfile stores the OCI tag, not the resolved version (a schema divergence from the official, now tracked as a follow-up). For now 9f6b0c8 guards the read so only a full-semver pin feeds `current`; cella's tag-only pin falls back to the resolved `wanted`, keeping current/wanted/latest all concrete.

File: crates/cella-cli/src/commands/outdated.rs [hunk 5]
```diff
@@ -41,216 +66,956 @@ pub struct OutdatedArgs {
...
+/// Resolve a user feature id to a versionable `(registry, repository, tag)`, or
+/// `None` if it is not a versionable OCI reference.
+///
+/// Returns `None` for local paths, tarball URLs, deprecated bare identifiers,
+/// and any reference that fails to parse — matching the official `getRef`
+/// returning `undefined`. The tag defaults to `"latest"` when absent.
+fn versionable_oci_ref(user_feature_id: &str) -> Option<(String, String, String)> {
+    use cella_features::{FeatureRef, NormalizedRef};
+
+    let parsed = FeatureRef::parse(user_feature_id).ok()?;
```
[the remark this answers is not in the document]
Comment (eve0415) [reply crates/cella-cli/src/commands/outdated.rs:312]: Fixed in 9f6b0c8 — digest-pinned refs are now skipped cleanly instead of listing tags for the mis-parsed repository. Full digest support (latest from the repo + the digest-resolved pinned version) is noted as a follow-up.
